python/tests/scripts/b0_chaining_prelim.metta

```diff
@@ -1,35 +1,42 @@
-; From matching to rule chaining
-
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; From matching to rule chaining:
+; It's possible to chain together matching queries
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Some knowledge base entries
 ; `:=` is a custom symbol
 ; Combinatory logic rules
 (:= (I $x) $x)
 (:= ((K $x) $y) $x)
 (:= (K $x $y) $x)
 (:= (S $x $y $z) ($x $z ($y $z)))
 
-; First step
+; A single matching step
 !(assertEqualToResult
   (match &self
     (:= (S K K x) $r)
     $r)
   ((K x (K x))))
 
-; Second step
+; Another single matching step
```

Comment:
It's not just "another single matching step". It should be underlined that this matching step takes the result of the previous matching step as input. "First" and "Second" provided an order for steps although were not that explicit regarding inputs/outputs. Now, the focus is on the fact that steps are single, but it is less important than the fact that these steps are manually chained.

Reply:
Oh, I missed that... but actually I don't understand how the result of the first step is used by the second step: if I comment out the "first step", the second one still works, because it is matched with the 3rd knowledge base entry:
```(:= (K $x $y) $x)```

Reply:
I mean that the first match produces `(K x (K x))`. This result is "manually" wrapped into the query - `(:= (K x (K x)) $r)`. This query is executed as the second "step" to produce the next result. You can explain explicitly that we take the result of the first query here and construct the next query, which performs the second step of reduction, if it is not clear enough from the code.

Reply:
And of course, the second step works if you comment out the first step, because the result of the first step is already in the second query.

Reply:
Got it, thx